app/src/main/java/com/example/todolist/ui/detailScreen/DetailedListViewModel.java

```diff
@@ -1,62 +1,76 @@
 package com.example.todolist.ui.detailScreen;
 
+import android.util.Log;
+
 import androidx.annotation.NonNull;
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
 import androidx.lifecycle.ViewModel;
 
 import com.example.todolist.dto.ListItem;
 import com.example.todolist.dto.ListItemComparator;
 import com.example.todolist.dto.ToDoList;
 import com.example.todolist.repositories.listItemsRepository.IListItemRepository;
-import com.example.todolist.repositories.listRepository.IListRepository;
 
 import java.util.ArrayList;
 import java.util.Collections;
+import java.util.concurrent.ExecutorService;
 
 public class DetailedListViewModel extends ViewModel {
+    public LiveData<ArrayList<ListItem>> listItemsLiveData;
+    private final MutableLiveData<ArrayList<ListItem>> listItemsMutableLiveData;
+
     private final int listID;
-    private String listHeader;
+    private final String listHeader;
+    public static final String TAG = "DetailsViewModel";
 
     private final IListItemRepository listItemRepository;
-    private final IListRepository listRepository;
 
-    private ArrayList<ListItem> listItems;
+    private final ExecutorService executor;
 
     public DetailedListViewModel(IListItemRepository listItemRepository,
-                                 IListRepository listRepository,
-                                 @NonNull ToDoList toDoList) {
+                                 @NonNull ToDoList toDoList,
+                                 ExecutorService executor) {
         this.listID = toDoList.getID();
         this.listHeader = toDoList.getHeader();
         this.listItemRepository = listItemRepository;
-        this.listRepository = listRepository;
-        listItems = new ArrayList<>();
+        this.executor = executor;
+        listItemsMutableLiveData = new MutableLiveData<>(new ArrayList<>());
+        listItemsLiveData = (LiveData<ArrayList<ListItem>>) listItemsMutableLiveData;
```

Comment:
Можешь также использовать следующий шаблон наименований мутабельных и иммутабельных объектов. Так часто делается.
```suggestion
        _listItemsLiveData = new MutableLiveData<>(new ArrayList<>());
        listItemsLiveData = (LiveData<ArrayList<ListItem>>) _listItemsLiveData;
```